### ISSUE TITLE
zbar crashes on QR code decoding

### ISSUE THREAD

**Issue by l3u** [mchehab/zbar#61]
I recently tried to decode a QR code of the chip tan stuff my bank does (out of curiosity what's encoded in it). Scanning such a QR code makes zbar crash. See the attached demo QR code:
![chipTanQr](https://user-images.githubusercontent.com/6770781/66905663-8dbdb180-f006-11e9-88df-642889ba6e31.png)

The content seems to be some binary data, using https://zxing.org/w/decode.jspx , I get:

    DK
    X@Ã\2'NrND;�]uywgp~[rì

Perhaps, zbar expects the QR code content not to be bytes? Anyway, would be cool if it wouldn't crash ;-)

**Comment by matheusmoreira** [comment]
I was able to reproduce the issue with `zbarimg 0.23`.

    double free or corruption (top)
    Aborted (core dumped)

However, I was able to decode the QR code with a `zbarimg` compiled from git.
The issue was fixed in commit 3b9c242a2975d3fdf4237fa2bdd05ef8aa89b6f1 which is newer than version 0.23.

---

This QR code contains Extended Channel Interpretation entry for `QR_ECI_GLI0` which signals that the data is encoded in CP437. It is followed by a binary representation of the following data:

    DK
    X@Ã\2'NrND;]uywgp~[rì
